### ISSUE TITLE
Free Private Repos

### ISSUE THREAD

**Issue by tuxy** [dear-github/dear-github#120]
Please make unlimited private repos like BitBucket and GitLab did. Thank You!

**Comment by joshmanders** [comment]
I don't think that's possible. Private repos are a big part of their revenue model. They could do a better job at how they charge.

**Comment by Qix-** [comment]
**Absolutely not.** I have no problem paying for private repos, because I have no problem monetarily supporting a company that does a good job. Same reason why I support Slack - their business model isn't greedy and makes sense for the prices they ask.

Pretty sure I'm speaking for most developers.

**Comment by garymoon** [comment]
I agree both that private repos shouldn't be free, but that they could do a better job charging.

I currently pay for a GitHub account that I don't really use (using bitbucket instead, but I dislike that it's free) because I wish to support GitHub, but have many more than 5 private repos, all of fairly small size. Charging by size or collaborators instead of number of repos would be great for me.

**Comment by tuxy** [comment]
I hear what you guys are saying, I don't mind having payed private repos too. But I think they should at least have 3-5 free ones for those who can't pay. Even 1 private repo would be a huge plus, for things like private dotfiles/configs you don't want seen public. So I'm not saying making it completely free, but something for everyone.

**Comment by Qix-** [comment]
Why not use gists?

**Comment by denji** [comment]
In gists there is a bug when you fork gist and make "Make Private" it is still possible to find based gist and link to private gist.

**Comment by tuxy** [comment]
Private Gists are useless compared to Private Repos :-)

**Comment by Qix-** [comment]
How are they useless compared to repos?

**Comment by Qix-** [comment]
They're perfect for dotfiles or config files

**Comment by joshmanders** [comment]
$7/mo for 5 private repos really isn't a lot, and it's pretty unreasonable to think that they should provide free private repos just because not everyone can afford them.

Are hosting providers required to offer free hosting because you can't afford the service?

**Comment by joschi** [comment]
> How are they useless compared to repos?

There's no authentication/authorization required to access "private" (e. g. only the URL isn't published) Gists. Private repositories on the other hand require valid credentials to access.

**Comment by aensley** [comment]
**I hope they never do this**. If you want free private repos, go to those other guys. GitHub is able to make a quality product because their customers (myself included) are willing to pay for it. I don't understand how free repos for open source isn't enough for some people. That's a heck of a lot. We are not entitled to free everything.

**Comment by Qix-** [comment]
@joschi that doesn't make them useless.

The fact you get free hosting for code isn't enough? $7 is too much? Saddening.

**Comment by joschi** [comment]
@Qix-
> @joschi that doesn't make them useless.

Let's just say it makes Gists less useful for dot-files, which may contain sensitive information you don't want to share with the world.

**Comment by Qix-** [comment]
@joschi less useful != useless. Also, gists' URLs are hashes. The chances of someone coming across them are slim.

As for private repositories, why not support the site that you use all the time by contributing $7 to keep them afloat? Is that not reasonable?

**Comment by aensley** [comment]
$7/month is less than a Netflix subscription...

**Comment by joschi** [comment]
@Qix-
> As for private repositories, why not support the site that you use all the time by contributing $7 to keep them afloat? Is that not reasonable?

If that was directed at me, I'm a paying customer. I was just addressing the question why Gists are less useful for hosting sensitive data than private repositories…

**Comment by Qix-** [comment]
You probably shouldn't be putting sensitive data on Github to begin with, honestly. :P

**Comment by brianjking** [comment]
I'd like to see perhaps 5 private repos for free (individuals only). If nothing else, I think it would really be nice to see private wiki/issues controls on existing repos or separate repos entirely.

But yeah, I definitely can say that $7 is really nothing and I'm a paying customer.

**Comment by looterz** [comment]
Just one private repo would be super appreciated, but I can see why they wouldn't as its a big part of their revenue model.

**Comment by Qix-** [comment]
One private repo would be acceptable I suppose. Obviously we have no idea how much that one repo affects their model, though.

**Comment by devonzara** [comment]
I agree, 1 or 2 private repos would be nice for non-commercial uses (eg. personal sites/blogs) or something, but I understand that it would likely affect their pricing model. I think it'd be something worth considering, but only if they can find a way to use it to their advantage without hurting their business.

Perhaps the free private repos could have some sort of limited functionality... Not to the point that it renders them nearly useless, but rather limit the functionality that would typically only be required by larger projects or commercially used repos - it would then act as an incentive to upgrade to a paid subscription.

**Comment by AlicanC** [comment]
Private repos should be free if only one person has access to them. This would cover:
- Dotfiles
- Personal blogs/websites
- Personal tools
- Projects that are WIP and yet to be open source-ready

I've actually realized that hosting my private repos on GitHub gives me zero benefit. Paying 12$/mo for nothing is stupid. It is my stupidity of course, I should host them on my VPS.

**Comment by joshmanders** [comment]
> Paying 12$/mo for nothing is stupid.

You're supporting GitHub. They gotta pay their bills somehow.

**Comment by AlicanC** [comment]
> > Paying 12$/mo for nothing is stupid.

> You're supporting GitHub. They gotta pay their bills somehow.

When I pay money for nothing I at least expect not having to fix nested quotes (like the one above) myself.

**Comment by joshmanders** [comment]
> When I pay money for nothing

Clearly if you want it so bad, it's not *nothing*.

**Comment by AlicanC** [comment]
> > When I pay money for nothing

> Clearly if you want it so bad, it's not nothing.

Your arguments are flawless. You've overturned my ignorance. Thank you, sir.

**Comment by joshmanders** [comment]
You're being childish and entitled.

**Comment by AlicanC** [comment]
Just don't go off topic to ask tips for a for-profit company so they can pay their bills, especially in a place where that company's lack of interest on needs of its customers are being discussed.

Seeing your comments here and on https://github.com/eslint/eslint/issues/5205 makes me think that you feel obliged to help GitHub. You are not. What a for-profit company offers for free is an advertisement for what they sell. What we are using here is a free sample, just like the ones we are handed in stores. Wikipedia is a service you should feel bad for not supporting, not GitHub.

Of course you could love GitHub so much that it makes you want to tip (and even ask others to tip) but this is really not the time for it. It is time for us to ground it and cut its allowance so it realizes it did something bad.

I am a person who tried a lot to move his company's repos to GitHub, despite it being more expensive. It is really embarrassing that they needed a public outrage in the first place and I'm not even sure they get the point. While people are trying to figure out how to migrate away, they introduce GitHub Importer. Now we get Issue and PR templates which is great, but it should be a half-day's work at most. What we get after is file uploading which solves no crucial problem and breaks if you click "Commit changes" before the upload completes.

I'm not sure if I'm being understood or given the middle finger.

**Comment by joshmanders** [comment]
> What a for-profit company offers for free is an advertisement for what they sell.

Private repos are not free.

> What we are using here is a free sample, just like the ones we are handed in stores.

Private repos are not free.

> Wikipedia is a service you should feel bad for not supporting, not GitHub.

That makes no sense, because GitHub is a for-profit company, nobody should support them?

> Of course you could love GitHub so much that it makes you want to tip (and even ask others to tip) but this is really not the time for it.

Nobody, especially GitHub are asking for tips.

> It is time for us to ground it and cut its allowance so it realizes it did something bad.

We did not make GitHub, GitHub made GitHub, we cannot "ground it" or "cut it's allowance". All we can do is leave the service.

> despite it being more expensive.

$7/mo for 5 private repos is not expensive.

> but it should be a half-day's work at most.

Clearly you've never worked on a big codebase or had ya know... Management to deal with. Bureaucratic bullshit isn't what I like, but who are we to tell GitHub how to run their business?

> I'm not sure if I'm being understood or given the middle finger.

You're being impatient, childish, and entitled.

**Comment by Qix-** [comment]
[Github has reduced their planset and has a single personal plan with unlimited private repositories](https://github.com/blog/2164-introducing-unlimited-private-repositories).

**Comment by ron-wolf** [comment]
> @Qix-
> 
> > @joschi that doesn't make them useless.
> 
> Let's just say it makes Gists less useful for dot-files, which may contain sensitive information you don't want to share with the world.

@joschi Dude, just add “`/.*`” to your `.gitignore`.

**Comment by joschi** [comment]
@ron-wolf Dude, you might want to read the whole thread again before replying. kthxbye.

**Comment by Qix-** [comment]
@joschi If your sensitive information is too sensitive to have in a secret gist with a 1 in 115792089237316195423570985008687907853269984665640564039457584007913129639935 (the actual number) chance of someone finding it, then you shouldn't be putting it on the internet.

:P

**Comment by joschi** [comment]
@Qix- Yeah, I mean do whatever you like and be happy with it. Peace! ☮️

**Comment by joshmanders** [comment]
I looks like I was talking to myself earlier.

**Comment by gingerbeardman** [comment]
There are now unlimited private repos on the Free tier! 

https://github.com/pricing
